verilog/analysis/checkers/numeric_format_string_style_rule_test.cc

```diff
@@ -0,0 +1,112 @@
+
+      // binary value
+      {"module test; initial $display(", {kToken, "\"%0b\""}, ", some_bin_value); endmodule"},
+      {"module test; initial $display(\"0b%0b\", some_binary_value); endmodule"},
+      {"module test; initial $display(\"'b%0b\", some_binary_value); endmodule"}, // also acceptable
+      {"module test; initial $display(\"Value: 0b%0b\", some_binary_value); endmodule"},
+      {"module test; initial $display(\"Value: 'b%0b\", some_binary_value); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: 0B%0b\""}, ", some_binary_value); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: 'B%0b\""}, ", some_binary_value); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: %0b\""}, ", some_binary_value); endmodule"},
+
+      // decimal value
+      {"module test; initial $display(\"%0d\", some_decimal_value); endmodule"}, // no prefix wanted
+      {"module test; initial $display(", {kToken, "\"0d%0d\""}, ", some_dec_value); endmodule"},
+      {"module test; initial $display(", {kToken, "\"'d%0d\""}, ", some_dec_value); endmodule"},
+      {"module test; initial $display(\"Value: %0d\", some_decimal_value); endmodule"}, // no prefix wanted
+      {"module test; initial $display(", {kToken, "\"Value: 0d%0d\""}, ", some_dec_value); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: 'd%0d\""}, ", some_dec_value); endmodule"},
+
+      // invalid prefix
+      {"module test; initial $display(", {kToken, "\"Value: 0b%0x\""}, ", some_hex_value); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: 0x%0b\""}, ", some_binary_value); endmodule"},
+
+      // TODO(fangism): Are those ok? Or just "0d" and "'d" are invalid prefixes for decimal values?
+      //{"module test; initial $display(", {kToken, "\"Value: 0x%0d\""}, ", some_dec_value); endmodule"},
+      //{"module test; initial $display(", {kToken, "\"Value: 0b%0d\""}, ", some_dec_value); endmodule"},
+
+      // multiple violations
+      {"module test; initial $display(", {kToken, "\"Value: 0x%0x %0x %0x\""}, ", hex1, hex2, hex3); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: %0x 'h%0x %0x\""}, ", hex1, hex2, hex3); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: %0x %0x %0x\""}, ", hex1, hex2, hex3); endmodule"},
+
+      {"module test; initial $display(", {kToken, "\"Value: 0b%0b %0b %0b\""}, ", bin1, bin2, bin3); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: %0b 'b%0b %0b\""}, ", bin1, bin2, bin3); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: %0b %0b %0b\""}, ",  bin1, bin2, bin3); endmodule"},
+
+      {"module test; initial $display(", {kToken, "\"Value: 0d%0d 'D%0d %0d\""}, ", dec1, dec2, dec3); endmodule"},
+      {"module test; initial $display(", {kToken, "\"Value: %0d 0D%0d 'd%0d\""}, ", dec1, dec2, dec3); endmodule"},
```

Comment:
These expected violations should be as narrow as possible, also allowing for multiple violations reported per string literal.